pastepwn/analyzers/tests/pastebinurlanalyzer_test.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+import unittest
+from unittest import mock
+
+from pastepwn.analyzers.pastebinurlanalyzer import PastebinURLAnalyzer
+
+class TestPastebinURLAnalyzer(unittest.TestCase):
+    def setUp(self):
+        self.analyzer = PastebinURLAnalyzer(None)
+        self.paste = mock.Mock()
+
+    def test_match_positive(self):
+        self.paste.body = "https://pastebin.com/xyz"
+        self.assertTrue(self.analyzer.match(self.paste))
+        self.paste.body = "https://pastebin.com/xyz/"
+        self.assertTrue(self.analyzer.match(self.paste))
+        self.paste.body = "http://pastebin.com/xyz"
+        self.assertTrue(self.analyzer.match(self.paste))
+        self.paste.body = "http://pastebin.com/xyz/"
+        self.assertTrue(self.analyzer.match(self.paste))
+        self.paste.body = "https://pastebin.com/xyz  "
+        self.assertTrue(self.analyzer.match(self.paste))
+
+    def test_match_negative(self):
+        self.paste.body = ""
+        self.assertFalse(self.analyzer.match(self.paste))
+        self.paste.body = None
+        self.assertFalse(self.analyzer.match(self.paste))
+        self.paste.body = "https://google.com/xyz"
+        self.assertFalse(self.analyzer.match(self.paste))
+        self.paste.body = "www.pastebin.com/xyz"
+        self.assertFalse(self.analyzer.match(self.paste))
```

Comment:
Currently the test fails here. I am not sure if it should fail. www as subdomain is valid because it redirects to pastebin.com.

I will get back to you later.

Reply:
I was under the impression that the url should begin with https or http but you are right it does redirect to pastebin.com. Let me know if you want treat the test is a positive match.

Reply:
Hi there, sorry for the late response. Busy days. I would say it's better to have this as a positive match, not a negative one. I would really appreciate if you could change the test accordingly (with and without `www` subdomain. Thank you!